clients/spark-extensions-base/src/test/java/org/projectnessie/spark/extensions/AbstractSparkSqlTest.java

```diff
@@ -718,4 +720,50 @@ void testValidCatalog() {
       spark.sessionState().catalogManager().setCurrentCatalog(catalog);
     }
   }
+
+  @ParameterizedTest
+  @ValueSource(strings = {refName, "main"})
+  void testCompaction(String branchName) throws NessieNotFoundException, NessieConflictException {
+    if (!branchName.equals("main")) {
+      assertThat(sql("CREATE BRANCH %s IN nessie FROM main", branchName))
+          .containsExactly(row("Branch", branchName, initialHashOfDefaultBranch));
+    }
+
+    sql("USE REFERENCE %s IN nessie", branchName);
+    sql("CREATE TABLE nessie.db.tbl (id int, name string)");
+    sql("INSERT INTO nessie.db.tbl select 23, \"test\"");
+    sql("INSERT INTO nessie.db.tbl select 24, \"test24\"");
+
+    String branchHash = api.getReference().refName(branchName).get().getHash();
+    assertThat(sql("CREATE BRANCH dev IN nessie FROM %s", branchName))
+        .containsExactly(row("Branch", "dev", branchHash));
+
+    List<Object[]> result =
+        sql(
+            "CALL nessie.system.rewrite_data_files(table => 'nessie.db.tbl', options => map('min-input-files','2'))");
+    // re-written files count is 2 and the added files count is 1
+    assertThat(result).hasSize(1).containsExactly(row(2, 1));
+
+    // commit hash should be modified due to compaction commit
+    String newHash = api.getReference().refName(branchName).get().getHash();
+    assertThat(newHash).isNotEqualTo(branchHash);
```

Comment:
Seems like it's worth to validate at least (part of) the commit message and the Iceberg operation type/name.

Reply:
checking the commit message now which will have the operation info ('replace' for compaction)